version: "3.9"
services:
  ml:
    build: ./fastapi_app
#    image: mar1a/msk23:base
#    deploy:
#      resources:
#        limits:
#          cpus: '0.80'
#          memory: 12g
    restart: on-failure
    ports:
      - "8000:8000"
    container_name: fastapi
    environment:
      PORT: 8000
      HOST: "postgres_db"
      OPENAI_API_KEY: ${OPENAI_API_KEY}
    volumes:
      - ./logs/fastapi/:/logs/
      - ./indexes/:/app/indexes/
      - ./fastapi_app/:/app/fastapi_app/ # mount the app folder to track changes
    networks:
      - mynetwork
    depends_on:
      postgres_db:
        condition: service_healthy

  postgres_db:
    image: "postgres:12"
    restart: always
    ports:
      - "5432:5432"
    environment:
      PORT: 5432
      POSTGRES_DB: "pgdatabase"
      POSTGRES_USER: "SECRET_USER"
      POSTGRES_PASSWORD: "SECRET_KEY"
    container_name: postgres_db
    volumes:
      - ./init-scripts:/docker-entrypoint-initdb.d
      - ./pgdata:/var/lib/postgresql/data
    networks:
      - mynetwork
    healthcheck:
        test: ["CMD", "pg_isready", "-q", "-d", "pgdatabase", "-U", "SECRET_USER"]
        interval: 10s
        timeout: 5s
        retries: 5

networks:
  mynetwork:
    driver: bridge
